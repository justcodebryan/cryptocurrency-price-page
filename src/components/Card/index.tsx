import classnames from 'classnames'
import { PropsWithChildren } from 'react'

import styles from './styles.module.scss'

type CardProps = PropsWithChildren

const Card = ({ children }: CardProps) => {
  const className = classnames([styles['card-wrapper'], styles['card']])

  return <div className={className}>{children}</div>
}

export default Card
