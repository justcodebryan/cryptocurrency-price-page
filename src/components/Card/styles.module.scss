.card {
  &-wrapper {
    width: 100%;
    // height: 160px;
    padding-bottom: 16px;
    border: 1px solid rgba($color: #000, $alpha: 35%);
    border-radius: 12px;
    box-sizing: border-box;
  }
}
