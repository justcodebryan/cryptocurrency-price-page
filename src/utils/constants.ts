import { NoopFn } from '@/types'

export const NOOP: NoopFn = () => void 0

export const DEFAULT_MODAL_CONTAINER_ID = 'modal-container'

export const DEFAULT_POLLING_INTERVAL = 10 * 1000
