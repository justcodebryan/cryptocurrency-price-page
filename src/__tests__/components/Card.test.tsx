import Card from '@/components/Card'
import { toJSON } from '@/utils/test-utils'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

describe('Card Test Suit', () => {
  test('Card Snapshot', () => {
    const component = renderer.create(<Card />)
    const tree = toJSON(component)
    expect(tree).toMatchSnapshot()
  })

  test('Should show children element', () => {
    const { container } = render(<Card>test</Card>)
    expect(container.querySelector('div')).toBeDefined()
    expect(screen.getByText('test')).toBeDefined()
  })
})
