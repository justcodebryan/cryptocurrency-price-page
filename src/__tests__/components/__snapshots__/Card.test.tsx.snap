// Vitest Snapshot v1

exports[`Card Test Suit > Card Snapshot 1`] = `
<div
  className="_card-wrapper_003db1 _card_003db1"
/>
`;
